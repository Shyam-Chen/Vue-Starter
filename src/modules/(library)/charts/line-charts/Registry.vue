<script lang="ts" setup>
import { reactive } from 'vue';

import Breadcrumbs from '~/components/Breadcrumbs.vue';
import Button from '~/components/Button.vue';
import Chart from '~/components/Chart.vue';

const flux = reactive({
  basicLineChart: {
    title: {
      text: '',
    },
    xAxis: {
      categories: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
    },
    yAxis: {
      title: {
        text: '',
      },
    },
    tooltip: {
      shared: true,
    },
    series: [
      {
        type: 'spline',
        name: '2021',
        data: [40, 68, 86, 74, 56, 60, 87],
        color: '#6366f1',
      },
      {
        type: 'spline',
        name: '2022',
        data: [65, 78, 66, 44, 56, 67, 75],
        color: '#ec4899',
      },
    ],
  },
  update() {
    const randomInt = (max = 99) => Math.floor(Math.random() * max) + 1;

    const idx_2021 = flux.basicLineChart.series.findIndex((serie) => serie.name === '2021');
    flux.basicLineChart.series[idx_2021].data = Array.from({ length: 7 }, () => randomInt());

    const idx_2022 = flux.basicLineChart.series.findIndex((serie) => serie.name === '2022');
    flux.basicLineChart.series[idx_2022].data = Array.from({ length: 7 }, () => randomInt());
  },
});
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Library', disabled: true },
      { text: 'Charts', disabled: true },
      { text: 'Line Charts', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Line Charts</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <Chart :options="flux.basicLineChart" />

    <Button class="mt-2" @click="flux.update">Update</Button>
  </div>
</template>
