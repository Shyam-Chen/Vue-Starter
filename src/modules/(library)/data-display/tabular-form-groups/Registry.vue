<script lang="ts" setup>
import { computed, reactive } from 'vue';
import { useSchema } from 'vue-formor';
import { string } from 'yup';

import Breadcrumbs from '~/components/Breadcrumbs.vue';

const state = reactive({
  listGroup: [
    {
      parent: 'O',
      children: [
        { firstField: 'O', secondField: '' },
        { firstField: '', secondField: 'O' },
        { firstField: 'O', secondField: 'O' },
        { firstField: '', secondField: '' },
      ],
    },
    {
      parent: '',
      children: [
        { firstField: '', secondField: 'O' },
        { firstField: 'O', secondField: '' },
        { firstField: '', secondField: '' },
        { firstField: 'O', secondField: 'O' },
      ],
    },
  ],
  errors: {} as Record<string, string>,
});

const schema = useSchema(
  [
    [
      computed(() => state.listGroup),
      (row: any) => [
        [computed(() => row.parent), computed(() => string().required())],
        [
          computed(() => row.children),
          (subRow: any) => [
            [computed(() => subRow.firstField), computed(() => string().required())],
            [computed(() => subRow.secondField), computed(() => string().required())],
          ],
        ],
      ],
    ],
  ],
  state,
);

schema.validate();
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Library', disabled: true },
      { text: 'Data Display', disabled: true },
      { text: 'Tabular Form Groups', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Tabular Form Groups</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic (Tabs + Tables)</div>

    <pre>{{ state.errors }}</pre>
  </div>
</template>
