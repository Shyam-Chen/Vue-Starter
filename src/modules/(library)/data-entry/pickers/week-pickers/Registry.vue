<script lang="ts" setup>
import { reactive } from 'vue';

import Breadcrumbs from '~/components/Breadcrumbs.vue';
import WeekPicker from '~/components/WeekPicker.vue';

const flux = reactive({
  weekPicker: [],
});
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Library', disabled: true },
      { text: 'Data Entry', disabled: true },
      { text: 'Pickers', disabled: true },
      { text: 'Week Pickers', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Week Pickers</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <WeekPicker v-model:value="flux.weekPicker" />
    </div>
  </div>
</template>
