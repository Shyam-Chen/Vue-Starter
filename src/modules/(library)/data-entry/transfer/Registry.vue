<script lang="ts" setup>
import Breadcrumbs from '~/components/Breadcrumbs.vue';
import Transfer from '~/components/Transfer.vue';
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Library', disabled: true },
      { text: 'Data Entry', disabled: true },
      { text: 'Transfer', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Transfer</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <Transfer />
    </div>
  </div>
</template>
