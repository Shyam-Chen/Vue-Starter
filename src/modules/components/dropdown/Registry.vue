<script lang="ts" setup>
import Breadcrumbs from '~/components/Breadcrumbs.vue';
import Dropdown from '~/components/Dropdown.vue';
import Button from '~/components/Button.vue';
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Platform', disabled: true },
      { text: 'Components', disabled: true },
      { text: 'Dropdown', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Dropdown</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>
    </div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-between">
      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>

      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>
    </div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>
    </div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-between">
      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>

      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>
    </div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <Dropdown :options="['Action', 'Another action', 'Something else here']">
        <Button>Button</Button>
      </Dropdown>
    </div>
  </div>
</template>
