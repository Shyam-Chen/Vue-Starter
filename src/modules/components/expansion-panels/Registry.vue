<script lang="ts" setup>
import Breadcrumbs from '~/components/Breadcrumbs.vue';
import ExpansionPanel from '~/components/ExpansionPanel.vue';
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Platform', disabled: true },
      { text: 'Components', disabled: true },
      { text: 'Expansion Panels', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Expansion Panels</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic</div>

    <div class="flex justify-center">
      <ExpansionPanel>
        <template #header>
          <div class="flex justify-between">
            <div>What is a "Composable"?</div>
            <div class="i-ic-baseline-arrow-drop-down w-6 h-6"></div>
          </div>
        </template>

        <template #content>
          <div class="space-y-4">
            <div>
              In the context of Vue applications, a "composable" is a function that leverages Vue's
              Composition API to encapsulate and reuse stateful logic.
            </div>

            <div>
              When building frontend applications, we often need to reuse logic for common tasks.
              For example, we may need to format dates in many places, so we extract a reusable
              function for that. This formatter function encapsulates stateless logic: it takes some
              input and immediately returns expected output. There are many libraries out there for
              reusing stateless logic - for example lodash and date-fns, which you may have heard
              of.
            </div>

            <div>
              By contrast, stateful logic involves managing state that changes over time. A simple
              example would be tracking the current position of the mouse on a page. In real world
              scenarios, it could also be more complex logic such as touch gestures or connection
              status to a database.
            </div>
          </div>
        </template>
      </ExpansionPanel>
    </div>
  </div>
</template>
