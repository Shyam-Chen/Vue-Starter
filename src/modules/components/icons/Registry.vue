<script lang="ts" setup>
import Breadcrumbs from '~/components/Breadcrumbs.vue';
</script>

<template>
  <Breadcrumbs
    :items="[
      { text: 'Platform', disabled: true },
      { text: 'Components', disabled: true },
      { text: 'Icons', disabled: true },
    ]"
    class="mb-4"
  />

  <div class="mb-4">
    <div class="text-3xl font-bold">Icons</div>
  </div>

  <div class="flex flex-col border p-4 mb-4">
    <div class="mb-2">Basic examples</div>

    <div class="flex justify-center">
      <div class="i-fa-bell w-3 h-3 text-blue-600"></div>
    </div>
  </div>
</template>
