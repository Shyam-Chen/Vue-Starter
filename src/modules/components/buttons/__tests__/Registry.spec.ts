import { test, expect } from 'vitest';
import { render } from '@testing-library/vue';

import Registry from '../Registry.vue';

test('Buttons', () => {
  const { html } = render(Registry);
  expect(html()).toMatchSnapshot();
});
