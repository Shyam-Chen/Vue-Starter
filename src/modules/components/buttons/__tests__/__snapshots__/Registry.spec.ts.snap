// Vitest Snapshot v1

exports[`Buttons 1`] = `
"<nav class=\\"rounded-md w-full mb-4\\">
  <ol class=\\"list-reset flex\\">
    <li><span class=\\"text-gray-500\\">Platform</span></li>
    <li>
      <div class=\\"text-gray-500 mx-2 flex items-center h-full\\">/</div>
    </li>
    <li><span class=\\"text-gray-500\\">Components</span></li>
    <li>
      <div class=\\"text-gray-500 mx-2 flex items-center h-full\\">/</div>
    </li>
    <li><span class=\\"text-gray-500\\">Buttons</span></li>
    <!--v-if-->
  </ol>
</nav>
<div class=\\"mb-4\\">
  <div class=\\"text-3xl font-bold\\">Buttons</div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Basic examples</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-white\\">
      <!--v-if-->Button
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Colors</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800\\">
      <!--v-if-->Primary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-purple-600 text-white hover:bg-purple-700 focus:bg-purple-700 active:bg-purple-800\\">
      <!--v-if-->Secondary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-green-500 text-white hover:bg-green-600 focus:bg-green-600 active:bg-green-700\\">
      <!--v-if-->Success
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-red-600 text-white hover:bg-red-700 focus:bg-red-700 active:bg-red-800\\">
      <!--v-if-->Danger
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-yellow-500 text-white hover:bg-yellow-600 focus:bg-yellow-600 active:bg-yellow-700\\">
      <!--v-if-->Warning
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-400 text-white hover:bg-blue-500 focus:bg-blue-500 active:bg-blue-600\\">
      <!--v-if-->Info
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Outline</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-blue-600 text-blue-600\\">Primary</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-purple-600 text-purple-600\\">Secondary</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-green-500 text-green-500\\">Success</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-red-600 text-red-600\\">Danger</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-yellow-500 text-yellow-500\\">Warning</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-blue-400 text-blue-400\\">Info</button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Rounded</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800 rounded-full\\">
      <!--v-if-->Primary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-purple-600 text-white hover:bg-purple-700 focus:bg-purple-700 active:bg-purple-800 rounded-full\\">
      <!--v-if-->Secondary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-green-500 text-white hover:bg-green-600 focus:bg-green-600 active:bg-green-700 rounded-full\\">
      <!--v-if-->Success
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-red-600 text-white hover:bg-red-700 focus:bg-red-700 active:bg-red-800 rounded-full\\">
      <!--v-if-->Danger
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-yellow-500 text-white hover:bg-yellow-600 focus:bg-yellow-600 active:bg-yellow-700 rounded-full\\">
      <!--v-if-->Warning
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-400 text-white hover:bg-blue-500 focus:bg-blue-500 active:bg-blue-600 rounded-full\\">
      <!--v-if-->Info
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Rounded outline</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-blue-600 text-blue-600 rounded-full\\">Primary</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-purple-600 text-purple-600 rounded-full\\">Secondary</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-green-500 text-green-500 rounded-full\\">Success</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-red-600 text-red-600 rounded-full\\">Danger</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-yellow-500 text-yellow-500 rounded-full\\">Warning</button><button type=\\"button\\" class=\\"inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-blue-400 text-blue-400 rounded-full\\">Info</button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Floating</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800\\">
      <div class=\\"w-3 mx-auto i-fa-download\\"></div>
    </button><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-purple-600 text-white hover:bg-purple-700 focus:bg-purple-700 active:bg-purple-800\\">
      <div class=\\"w-3 mx-auto i-fa-share-square-o\\"></div>
    </button><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-green-500 text-white hover:bg-green-600 focus:bg-green-600 active:bg-green-700\\">
      <div class=\\"w-3 mx-auto i-fa-plus\\"></div>
    </button><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-red-600 text-white hover:bg-red-700 focus:bg-red-700 active:bg-red-800\\">
      <div class=\\"w-3 mx-auto i-fa-minus\\"></div>
    </button><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-yellow-500 text-white hover:bg-yellow-600 focus:bg-yellow-600 active:bg-yellow-700\\">
      <div class=\\"w-3 mx-auto i-fa-exclamation\\"></div>
    </button><button type=\\"button\\" class=\\"inline-block rounded-full leading-normal uppercase shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out w-9 h-9 bg-blue-400 text-white hover:bg-blue-500 focus:bg-blue-500 active:bg-blue-600\\">
      <div class=\\"w-3 mx-auto i-fa-question\\"></div>
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Disabled state</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800 pointer-events-none opacity-60\\">
      <!--v-if-->Primary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-purple-600 text-white hover:bg-purple-700 focus:bg-purple-700 active:bg-purple-800 pointer-events-none opacity-60\\">
      <!--v-if-->Secondary
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-green-500 text-white hover:bg-green-600 focus:bg-green-600 active:bg-green-700 pointer-events-none opacity-60\\">
      <!--v-if-->Success
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-red-600 text-white hover:bg-red-700 focus:bg-red-700 active:bg-red-800 pointer-events-none opacity-60\\">
      <!--v-if-->Danger
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-yellow-500 text-white hover:bg-yellow-600 focus:bg-yellow-600 active:bg-yellow-700 pointer-events-none opacity-60\\">
      <!--v-if-->Warning
    </button><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-400 text-white hover:bg-blue-500 focus:bg-blue-500 active:bg-blue-600 pointer-events-none opacity-60\\">
      <!--v-if-->Info
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">With icon</div>
  <div class=\\"flex space-x-2 justify-center\\"><button type=\\"button\\" class=\\"inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800 flex items-center\\">
      <div class=\\"w-3 mr-2 i-fa-download\\"></div>Download
    </button></div>
</div>
<div class=\\"flex flex-col border p-4 mb-4\\">
  <div class=\\"mb-2\\">Block buttons</div>
  <div><button class=\\"w-full mb-2 inline-block px-6 py-2.5 font-medium text-xs leading-tight uppercase rounded shadow-md hover:shadow-lg focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg transition duration-150 ease-in-out bg-blue-600 text-white hover:bg-blue-700 focus:bg-blue-700 active:bg-blue-800 w-full mb-2\\" type=\\"button\\">
      <!--v-if-->Button
    </button><button class=\\"w-full inline-block px-6 py-2 bg-white border-2 font-medium text-xs leading-tight uppercase rounded hover:bg-black hover:bg-opacity-5 focus:outline-none focus:ring-0 transition duration-150 ease-in-out border-blue-600 text-blue-600 w-full\\" type=\\"button\\">Button</button></div>
</div>"
`;
